x-pack/plugins/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts

```diff
@@ -51,19 +56,32 @@ export async function createAnomalyDetectionJobs({
     );
 
     const apmMetricIndex = indices.metric;
-    const responses = await Promise.all(
-      uniqueMlJobEnvs.map((environment) =>
-        createAnomalyDetectionJob({ mlClient, environment, apmMetricIndex })
-      )
-    );
+    const responses = [];
+    const failedJobs = [];
+    // Avoid the creation of multiple ml jobs in parallel
+    // https://github.com/elastic/elasticsearch/issues/36271
+    for (const environment of uniqueMlJobEnvs) {
+      try {
+        responses.push(
+          await createAnomalyDetectionJob({
+            mlClient,
+            esClient,
+            environment,
+            apmMetricIndex,
+          })
+        );
+      } catch ({ id, error }) {
```

Comment:
What happens if an unexpected error is thrown, eg `new Error('Boo!')` ? In that `id` and `error` is not available

Reply:
I get your point here, what would you recommend?
as of now we will be getting
```
An error occurred while creating ML jobs: [{}]
```

Maybe using the `environment` we were using in first place?

Reply:
If you know the error instance name, you can do like:

```suggestion
      } catch (e) {
      if(!e instanceof MLError) {
       throw e;
      } 
      failedJobs.push({ id: e.id, error: e.error });
        
```

Reply:
Then you can do

```suggestion
      } catch (e) {
      if(!e.id || !e.error) {
       throw e;
      } 
      failedJobs.push({ id: e.id, error: e.error });
        
```

Reply:
This is marked as resolved but it still assumes a an object to be thrown.

Reply:
Oh, nvm. I was looking at an old commit